main.js

```diff
@@ -15,45 +17,72 @@ lockButton.addEventListener('click', function(event) {
     }
 });
 
-newPaletteButtonSection.addEventListener('click', function(event) {
-    if(event.target.classList.contains('button-box-l') || event.target.id === 'new-palette' || event.target.classList.contains('button-box-r')) {
+buttonSection.addEventListener('click', function(event) {
+    if(event.target.classList.contains('new-palette')) {
         getNewHexes();
     }
 });
 
 function getNewHexes() {
-    currentHexes = [];
+    var oldHexes = currentPalette
+    currentPalette = [];
     var newColor;
     for(i = 0; i < mainColorBoxes.length; i++) {
         var thisColorBoxLock = mainColorBoxes[i].firstElementChild.firstElementChild
         if(thisColorBoxLock.classList.contains('unlocked')) {
             newColor = getRandomHex().toUpperCase();
             mainColorBoxes[i].firstElementChild.style.backgroundColor = `#${newColor}`;
             mainColorBoxes[i].lastElementChild.innerText = `#${newColor}`;
+            currentPalette.push(newColor);
+        } else { 
+            currentPalette.push(oldHexes[i])
         }
     }
 }
 
 function lockToggle(event) {
     if (event.getAttribute('src') === './assets/unlocked.png') {
         event.src = './assets/locked.png';
-        toggleLockClass(event);
     } else {
-        event.src = './assets/unlocked.png';
-        toggleLockClass(event);
-    }
-}
-
-function toggleLockClass(element) {
-    if(element.classList.contains('locked')) {
-        element.classList.remove('locked');
-        element.classList.add('unlocked');
-    } else {
-        element.classList.add('locked');
-        element.classList.remove('unlocked');
+        event.src = './assets/unlocked.png'; 
     }
+    event.classList.toggle('unlocked')
 }
 
 function getRandomHex() {
     return (Math.floor(Math.random() * 16777216).toString(16).padStart(6, 0));
 }
+
+buttonSection.addEventListener('click', function(event) {
+    if (event.target.classList.contains('save-palette')) {
+        savePalette();
+    }
+})
```

Comment:
Can we please move this up with the other eventlisteners?  I'm not sure what the actual best practices are, but I def got negative feedback on my last project that the eventlisteners were not grouped together towards the top :)